src/lib/mongoBackend/MongoCommonUpdate.cpp

```diff
@@ -1172,11 +1170,9 @@ static bool processContextAttributeVector (ContextElement*
     for (unsigned int ix = 0; ix < ceP->contextAttributeVector.size(); ++ix) {
 
         ContextAttribute* targetAttr = ceP->contextAttributeVector.get(ix);
-
         /* No matter if success or fail, we have to include the attribute in the response */
         ContextAttribute* ca = new ContextAttribute(targetAttr->name, targetAttr->type);
-        setResponseMetadata(targetAttr, ca);
-        cerP->contextElement.contextAttributeVector.push_back(ca);
+        setResponseMetadata(targetAttr, ca);        
 
```

Comment:
Avoiding duplicating the same line all the time would be great, but I think I cannot, as in some of the "logic branches" the `found` field in `ca` could need to be modified.

```
        if (strcasecmp(action.c_str(), "update") == 0) {
            if (updateAttribute(attrs, newAttrs, targetAttr, actualUpdate)) {
                ...
            }
            else {
                ...            
                ca->found = false;
            }
```

Knowing that, if you think it can be improved in any way, please tell me (I don't like the way it is right now... probably it is a bette way of re-structuring the logic).